.github/workflows/main.yml

```diff
@@ -18,34 +18,49 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
 
+env:
+  COMMIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
```

Comment:
`github.sha` in a pull request is the SHA of the merge commit GitHub creates, not the head of the pull request. We want to use the latter so that, e.g., we can use the hash of the latest commit to perform deploys.